src/netstandard/src/ApiCompatBaseline.monoandroid.txt

```diff
@@ -384,6 +384,10 @@ CannotRemoveBaseTypeOrInterface : Type 'System.IO.FileStream' does not implement
 CannotMakeMemberNonVirtual : Member 'System.IO.FileStream.Name' is non-virtual in the implementation but is virtual in the contract.
 MembersMustExist : Member 'System.IO.FileStream.DisposeAsync()' does not exist in the implementation but it does exist in the contract.
 CannotMakeMemberNonVirtual : Member 'System.IO.FileStream.Name.get()' is non-virtual in the implementation but is virtual in the contract.
+MembersMustExist : Member 'System.IO.FileStream.Read(System.Span<System.Byte>)' does not exist in the implementation but it does exist in the contract.
```

Comment:
@ericstj I'm a bit surprised by these entries from API Compat. Why does it enforce that overrides in the reference assembly exist in the implementation? Missing/additional overrides are legal in IL.

/cc @jkotas

Reply:
FileStream isn't sealed, so if someone extended it then a call to `base.Read(span)` needs to target FileStream's version, instead of Stream's version.

So it is actually an important compile differentiating change (on unsealed types).

Reply:
Sure, but if I'm compiling against .NET Standard, the override simply adds a layer of indirection. The call is recorded against FileStream's version. It's valid in IL to call a method through a derived type, so in .NET Core the call still ends being dispatched to Stream's version (assuming FileStream didn't provide the override).

Reply:
I agree - the tooling is more strict than it needs to be.

Reply:
I suppose in that sense our reference assemblies should say they override everything if the type isn't sealed, just in case...  I'd prefer it if the compiler just did the more conservative thing.

APICompat needs work, it's essentially just a code-record of all these conversations.  File an issue, please. https://github.com/dotnet/arcade/issues

Reply:
Is changing the compiler behaviour here a feasible option? It sounds like it would be a good idea more broadly. In particular, the [breaking changes doc](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md#members) claims that overrides can be added or removed at will, which could lead to very subtle breaking changes with the current compiler behaviour.

(I suspect there'll be subtle reasons why it can't be done, but I'd like it to be considered seriously if that wasn't already happening.)

Reply:
> Is changing the compiler behaviour here a feasible option?

We've talked about this before. AFAIR the conclusion that it wasn't acceptable. However, as @ericstj said, we could always emit overrides in reference assemblies. The downside is potential bloat in metadata (especially for hierharchies with many virtuals, such as WinForms and WPF). On the other hand, I'm not aware of any cases where today's behavior has caused a real bug. I'm sure cases exist, but it seems extremely rare.

Reply:
@ericstj 

> APICompat needs work, it's essentially just a code-record of all these conversations. File an issue, please. https://github.com/dotnet/arcade/issues

Filed https://github.com/dotnet/arcade/issues/2570.